kpops/components/base_components/models/from_section.py

```diff
@@ -38,19 +42,39 @@ class Config(DescConfig):
 
     @root_validator
     def extra_topic_role(cls, values: dict) -> dict:
-        """Ensure that cls.role is used correctly"""
-        is_extra_topic = values["type"] in (
-            InputTopicTypes.EXTRA,
-            InputTopicTypes.EXTRA_PATTERN,
-        )
-        if is_extra_topic and not values.get("role"):
-            raise ValueError(
-                "If you define an extra input component, extra input topic, or extra input pattern, you have to define a role."
-            )
-        if not is_extra_topic and values.get("role"):
-            raise ValueError(
-                "If you do not define an input component, input topic, or input pattern, the role is unnecessary."
-            )
+        """Ensure that cls.role is used correctly, assign type if needed"""
+        type_ = values.get("type", None)
+        role = values.get("role", None) is not None
+        # Assign type
+        match type_, role:
+            case None, False:
+                values["type"] = InputTopicTypes.INPUT
+                return values
+            case None, True:
+                values["type"] = InputTopicTypes.EXTRA
+                return values
+            case InputTopicTypes.PATTERN, False:
+                values["type"] = InputTopicTypes.INPUT_PATTERN
+                return values
+            case InputTopicTypes.PATTERN, True:
+                values["type"] = InputTopicTypes.EXTRA_PATTERN
+                return values
```

Comment:
just to confirm: the check below is unnecessary in this case, right? hence the early return you're doing here

Reply:
yeah, if `pattern` and `role`, `type=extra-pattern`

Reply:
The check below exists, because right now the user is free to set the type to `extra-pattern` or `extra`